### PR TITLE
PHP7 for sabre-dav 3.0

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@evert all green on php7 https://travis-ci.org/fruux/sabre-dav/builds/100782445 - requires https://github.com/fruux/sabre-vobject/pull/287

**Comment by DeepDiver1975** [comment]
@evert as soon as there is vobject 3.5 available I'll update this PR - thanks for your quick support on this!

**Comment by DeepDiver1975** [comment]
@evert how do we move on with this? THX

**Comment by evert** [comment]
I need to release sabre/vobject 3.5 first. After that this PR should probably be squashed and get a better title. After that I can merge this and release sabre/dav 3.0.7.

Should have time for all of that this week, but it's an annoying multi-step process and entails preparing blogposts and stuff. I'll probably do a set of releases for vobject 4, sabre/dav 3.1 first though as there's a few bugfixes there.

**Comment by DeepDiver1975** [comment]
> Should have time for all of that this week, but it's an annoying multi-step process and entails preparing blogposts and stuff. I'll probably do a set of releases for vobject 4, sabre/dav 3.1 first though as there's a few bugfixes there.

Awesome - Thanks a lot!

**Comment by DeepDiver1975** [comment]
> After that this PR should probably be squashed and get a better title.

I'll take care of that once 3.5 is released - THX

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     ],
     "require": {
         "php": ">=5.4.1",
-        "sabre/vobject": ">=3.3.4,<4",
+        "sabre/vobject": "3.5.x-dev",
```
Comment (evert) [inline composer.json]: When you update this PR, please change this dependency to `^3.3.4`.

Comment (DeepDiver1975) [inline composer.json]: 3.3.4 ? But we want 3.5.0 to get php7 support - right?

Comment (evert) [inline composer.json]: `^3.3.4` is identical to `>=3.3.4,<4` so it includes `3.5.0`. It's a good idea with dependencies to catch an as wide as possible net and only increase the minimum version when it's absolutely needed.

**Comment by DeepDiver1975** [comment]
@evert composer.json updated as requested and commit history clean up as well

**Comment by evert** [comment]
Awesome. Will merge this as soon as travis says it's ok.

**Comment by DeepDiver1975** [comment]
Thx

**Pull request merged**